include/deal.II/base/quadrature_point_data.h

```diff
@@ -328,7 +328,7 @@ namespace parallel
     {
     public:
       DEAL_II_STATIC_ASSERT(std::is_base_of<TransferableQuadraturePointData, DataType>::value,
-                            "User's DataType class should be derived from QPData");
+                            "User's DataType class should be derived from QPData")
```

Comment:
What about the one and only true C style
```
#if defined(DEAL_II_WITH_CXX11)
#define DEAL_II_STATIC_ASSERT(...) static_assert(__VA_ARGS__)
#else
#define DEAL_II_STATIC_ASSERT(...) do { } while(0)
#endif
```
With that the current syntax
```
DEAL_II_STATIC_ASSERT(...);
```
continues to work and we should see no warning.